test/EVMHost.cpp

```diff
@@ -357,7 +358,8 @@ evmc::Result EVMHost::call(evmc_message const& _message) noexcept
 	auto& destination = accounts[message.recipient];
 	if (message.kind == EVMC_CREATE || message.kind == EVMC_CREATE2)
 		// Mark account as created if it is a CREATE or CREATE2 call
-		newlyCreatedAccounts.emplace(message.recipient);
+		// TODO: Should we roll changes back on failure like we do for `accounts`?
+		m_newlyCreatedAccounts.emplace(message.recipient);
```

Comment:
Note that below (and also in a few other places in this function) we roll back changes to `accounts` with `accounts = stateBackup` when the result is a failure. I suspect we should be doing the same with the list of newly created accounts, because otherwise it may contain accounts that ended up not being actually created.

I don't think it can cause any wrong behavior in practice though (we only use the list to decide which accounts to remove and `accounts.erase()` will ignore non-existent ones) so a TODO note may be enough for now.

Reply:
> I don't think it can cause any wrong behavior in practice though (we only use the list to decide which accounts to remove and `accounts.erase()` will ignore non-existent ones) so a TODO note may be enough for now.

By `accounts.erase() will ignore non-existent one` do you mean the case when `m_newlyCreatedAccounts` has an account that does not exists in `accounts` because of the roll back? So, it will enter in this `if` https://github.com/ethereum/solidity/blob/006cd49313f3f64663fe0c4e09b0dcbb20f8e2be/test/EVMHost.cpp#L202 since the address will exists in `m_newlyCreatedAccounts` but then the `accounts.erase(address)` will do nothing, right?

Reply:
Yeah, that case exactly.

> Does this scenario can really happen?
One of the situations where a rollback happens is when the contract reverts due to insuffcient funds to transfer. I think it should be possible to trigger the problem by using `new` and giving it too high `value` so that it reverts due to lack of funds.

> Do we remove an account after a failure?

Yeah. `accounts = stateBackup` is what effectively removes it.

Reply:
I see. So yeah, I agree that the `TODO` is fine for now.